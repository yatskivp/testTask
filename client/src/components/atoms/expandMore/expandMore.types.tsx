import { IconButtonProps } from '@mui/material';

export type ExpandMoreProps = IconButtonProps & { expand: boolean };
