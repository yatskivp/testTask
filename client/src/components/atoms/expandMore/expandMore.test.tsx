import { render } from '@testing-library/react';

import { ExpandMore } from './expandMore';

describe('ExpandMore', () => {
  it('should match correct snapshot in expanded state', () => {
    const { container } = render(<ExpandMore expand={true} />);
    expect(container).toMatchSnapshot();
  });

  it('should match correct snapshot in collapsed state', () => {
    const { container } = render(<ExpandMore expand={false} />);
    expect(container).toMatchSnapshot();
  });
});
