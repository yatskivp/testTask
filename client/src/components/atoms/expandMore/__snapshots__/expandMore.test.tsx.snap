// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandMore should match correct snapshot in collapsed state 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-6ts3b8-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span>
      <svg
        height="16px"
        id="Layer_1"
        version="1.1"
        viewBox="0 0 32 32"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.221,7.206l9.585,9.585c0.879,0.879,0.879,2.317,0,3.195l-0.8,0.801c-0.877,0.878-2.316,0.878-3.194,0  l-7.315-7.315l-7.315,7.315c-0.878,0.878-2.317,0.878-3.194,0l-0.8-0.801c-0.879-0.878-0.879-2.316,0-3.195l9.587-9.585  c0.471-0.472,1.103-0.682,1.723-0.647C17.115,6.524,17.748,6.734,18.221,7.206z"
          fill="#fff"
        />
      </svg>
    </span>
  </button>
</div>
`;

exports[`ExpandMore should match correct snapshot in expanded state 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-pr9tn1-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span>
      <svg
        height="16px"
        id="Layer_1"
        version="1.1"
        viewBox="0 0 32 32"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.221,7.206l9.585,9.585c0.879,0.879,0.879,2.317,0,3.195l-0.8,0.801c-0.877,0.878-2.316,0.878-3.194,0  l-7.315-7.315l-7.315,7.315c-0.878,0.878-2.317,0.878-3.194,0l-0.8-0.801c-0.879-0.878-0.879-2.316,0-3.195l9.587-9.585  c0.471-0.472,1.103-0.682,1.723-0.647C17.115,6.524,17.748,6.734,18.221,7.206z"
          fill="#fff"
        />
      </svg>
    </span>
  </button>
</div>
`;
